test/fail_compilation/test15191.d

```diff
@@ -61,10 +60,23 @@ ref int*[1] identityArr(return ref scope int*[1] x)
 	return x;
 }
 
-int* sliceOfRefEscape()
+int*[] sliceOfRefEscape()
 {
 	int stackVar = 0xFF;
 	scope int*[1] x = [&stackVar];
 	int*[] y = identityArr(x)[];
-	return y[0];
+	return identityArr(x)[];
+}
```

Comment:
Why is this test modified? The `y` declaration becomes useless that way.

Reply:
I needed test coverage for `checkTransitiveScope` inside `checkReturnEscape`. `y` is not relevant, the important part is that `identityArr(x)[];` is rejected.

Reply:
Then just delete the declaration.

Reply:
It needs to be assigned to something for `checkAssignEscape`, but I can `cast(void)` to explicitly discard it